File: my-folder/1046-max-consecutive-ones-iii/solution.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        L, R = 0, 0
        longest_seq = 0
        count_zeros = 0
        while R < len(nums):
            if nums[R] == 0:
                count_zeros += 1
            
            while count_zeros > k:
                if nums[L] == 0:
                    count_zeros -= 1
                L += 1
            
            longest_seq = max(longest_seq, R-L + 1)
            R +=  1
        return longest_seq


            

        

File: my-folder/1046-max-consecutive-ones-iii/solution.java
class Solution {
    public int longestOnes(int[] nums, int k) {
        // sliding window approach 
        int totalZeros = 0;
        int left = 0, right = 0;
        int maxLen = 0;

        while(right < nums.length){
            if(nums[right] == 0){
                totalZeros += 1;
            }

            while(totalZeros > k){
                if(nums[left] == 0){
                    totalZeros--;
                }
                left += 1;
            }

            maxLen = Math.max(maxLen, right - left + 1);
            right += 1;
        }

        return maxLen;
    }
}
